### PR TITLE
Add aws_account_alias to aws_service

### PR DESCRIPTION
**Pull request by javiersanchz**
|Related issue|
|---|
| #15840  |

## Description

The `aws_account_alias` parameter was added to `aws_service` because it was not yet implemented in the code, but it was present in the documentation.

The changes were verified, modified accordingly, and `unit_tests` were run. Additionally, the documentation was reviewed to ensure it did not require any updates.

## Configuration options

Enabled `wazuh_modules.debug=2` and `/var/ossec/etc/ossec.conf`:

```console
  <wodle name="aws-s3">
    <disabled>no</disabled>
    <interval>1m</interval>
    <run_on_start>yes</run_on_start>
    
    <service type="cloudwatchlogs">a
      <aws_profile>default</aws_profile>
      <aws_log_groups>wazuh-cloudwatchlogs-integration-tests</aws_log_groups>
      <aws_account_alias>test-feature-15840</aws_account_alias>
      <regions>us-east-1</regions>
    </service>
  </wodle>
```

## Logs

```console
2024/02/09 12:43:23 wazuh-modulesd:aws-s3[19177] wm_aws.c:703 at wm_aws_run_service(): DEBUG: Service: cloudwatchlogs(test-feature-15840)  -  OUTPUT: DEBUG: +++ Debug mode on - Level: 2
DEBUG: +++ Getting alerts from "us-east-1" region.
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: only logs: None
DEBUG: Getting log streams for "wazuh-cloudwatchlogs-integration-tests" log group
DEBUG: Found "wazuh-cloudwatchlogs-integration-tests" log stream in wazuh-cloudwatchlogs-integration-tests
DEBUG: Getting data from DB for log stream "wazuh-cloudwatchlogs-integration-tests" in log group "wazuh-cloudwatchlogs-integration-tests"
DEBUG: Token: "f/38076214461433656xxxxxs", start_time: "1707350400000", end_time: "1707350400005"
DEBUG: Getting CloudWatch logs from log stream "wazuh-cloudwatchlogs-integration-tests" in log group "wazuh-cloudwatchlogs-integration-tests" using token "f/3807621446143xxxxx/s", start_time "1707350400006" and end_time "None"
DEBUG: +++ There are no new events in the "wazuh-cloudwatchlogs-integration-tests" group
DEBUG: Saving data for log group "wazuh-cloudwatchlogs-integration-tests" and log stream "wazuh-cloudwatchlogs-integration-tests".
DEBUG: The saved values are "{'token': 'f/38076214461433xxxxxxs', 'start_time': 1707350400000, 'end_time': 1707350400006}"
DEBUG: Some data already exists on DB for that key. Updating their values...
DEBUG: Purging the BD
DEBUG: Getting log streams for "wazuh-cloudwatchlogs-integration-tests" log group
DEBUG: Found "wazuh-cloudwatchlogs-integration-tests" log stream in wazuh-cloudwatchlogs-integration-tests
DEBUG: committing changes and closing the DB
2024/02/09 12:43:23 wazuh-modulesd:aws-s3[19177] wm_aws.c:80 at wm_aws_main(): DEBUG: Sleeping until: 2024/02/09
```
Service with the `account_alias` **test-feature-15840** in the `ossec.log` output: 

```
DEBUG: Service: cloudwatchlogs(test-feature-15840)
```

## Tests

<details>

<summary>Test_aws_service</summary>

```console
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.9.16, pytest-7.3.1, pluggy-0.13.1 -- /home/wazuh/venv/unittest-env/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.9.16', 'Platform': 'Linux-6.2.0-37-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.3.1', 'pluggy': '0.13.1'}, 'Plugins': {'metadata': '3.0.0', 'aiohttp': '1.0.4', 'html': '2.1.1', 'trio': '0.7.0', 'asyncio': '0.18.1', 'tavern': '1.23.5'}}
rootdir: /home/wazuh/Git/wazuh/wodles/aws/tests
configfile: pytest.ini
plugins: metadata-3.0.0, aiohttp-1.0.4, html-2.1.1, trio-0.7.0, asyncio-0.18.1, tavern-1.23.5
asyncio: mode=auto
collected 7 items                                                                                                                                                                                          

wodles/aws/tests/test_aws_service.py::test_aws_service_initializes_properly PASSED                                                                                                                   [ 14%]
wodles/aws/tests/test_aws_service.py::test_aws_service_check_region PASSED                                                                                                                           [ 28%]
wodles/aws/tests/test_aws_service.py::test_aws_service_get_last_log_date PASSED                                                                                                                      [ 42%]
wodles/aws/tests/test_aws_service.py::test_aws_service_format_message PASSED                                                                                                                         [ 57%]
wodles/aws/tests/test_aws_service.py::test_aws_service_get_alert_msg[event0] PASSED                                                                                                                  [ 71%]
wodles/aws/tests/test_aws_service.py::test_aws_service_get_alert_msg[event1] PASSED                                                                                                                  [ 85%]
wodles/aws/tests/test_aws_service.py::test_aws_service_get_alert_msg[None] PASSED                                                                                                                    [100%]

============================================================================================ 7 passed in 0.20s =============================================================================================
```

</details>

<details>

<summary>Test_aws</summary>

```console
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.9.16, pytest-7.3.1, pluggy-0.13.1
rootdir: /home/wazuh/Git/wazuh/wodles/aws/tests
configfile: pytest.ini
plugins: metadata-3.0.0, aiohttp-1.0.4, html-2.1.1, trio-0.7.0, asyncio-0.18.1, tavern-1.23.5
asyncio: mode=auto
collected 642 items                                                                                                                                                                                               

wodles/aws/tests/test_aws_bucket.py ....................................................................................................................................................................... [ 26%]
..........................................                                                                                                                                                                  [ 32%]
wodles/aws/tests/test_aws_s3.py ..................                                                                                                                                                          [ 35%]
wodles/aws/tests/test_aws_service.py .......                                                                                                                                                                [ 36%]
wodles/aws/tests/test_cloudtrail.py ..                                                                                                                                                                      [ 36%]
wodles/aws/tests/test_cloudwatchlogs.py .....................................................                                                                                                               [ 45%]
wodles/aws/tests/test_config.py ..............................................................................                                                                                              [ 57%]
wodles/aws/tests/test_guardduty.py .................                                                                                                                                                        [ 59%]
wodles/aws/tests/test_inspector.py ......                                                                                                                                                                   [ 60%]
wodles/aws/tests/test_load_balancers.py ............                                                                                                                                                        [ 62%]
wodles/aws/tests/test_s3_log_handler.py ................                                                                                                                                                    [ 65%]
wodles/aws/tests/test_server_access.py .................................                                                                                                                                    [ 70%]
wodles/aws/tests/test_sqs_message_processor.py ........                                                                                                                                                     [ 71%]
wodles/aws/tests/test_sqs_queue.py .......                                                                                                                                                                  [ 72%]
wodles/aws/tests/test_tools.py ..................................                                                                                                                                           [ 77%]
wodles/aws/tests/test_umbrella.py ......                                                                                                                                                                    [ 78%]
wodles/aws/tests/test_vpcflow.py ...........................                                                                                                                                                [ 83%]
wodles/aws/tests/test_waf.py .......                                                                                                                                                                        [ 84%]
wodles/aws/tests/test_wazuh_integration.py ......................................................................................................                                                           [100%]

=============================================================================================== 642 passed in 2.60s ===============================================================================================
```

</details>

### CODE REVIEW

**Review by GGP1 — CHANGES_REQUESTED** [review]

**Review by GGP1 — CHANGES_REQUESTED** [review]
The account alias options is being added to the AWS services classes, but I can't find any indication of whether it's being added to the logs like buckets do [here](https://github.com/wazuh/wazuh/blob/076d22b3e3f0f8bed7a5fbeef48cb5199adaed90/wodles/aws/buckets_s3/aws_bucket.py#L325-L329). Please upload evidence that the alias appears in the logs as well.

File: wodles/aws/services/cloudwatchlogs.py [hunk 1]
```diff
@@ -64,7 +66,7 @@ class AWSCloudWatchLogs(aws_service.AWSService):
     """
 
     def __init__(self, reparse, access_key, secret_key, profile,
-                 iam_role_arn, only_logs_after, region, aws_log_groups,
+                 iam_role_arn, only_logs_after, account_alias ,region, aws_log_groups,
```
Comment (GGP1) [inline wodles/aws/services/cloudwatchlogs.py:69]:
```suggestion
                 iam_role_arn, only_logs_after, account_alias, region, aws_log_groups,
```

File: wodles/aws/services/cloudwatchlogs.py [hunk 2]
```diff
@@ -137,8 +139,8 @@ def __init__(self, reparse, access_key, secret_key, profile,
         aws_service.AWSService.__init__(self, db_table_name='cloudwatch_logs', service_name='cloudwatchlogs',
                                         reparse=reparse, access_key=access_key, secret_key=secret_key,
                                         profile=profile, iam_role_arn=iam_role_arn,
-                                        only_logs_after=only_logs_after, region=region, discard_field=discard_field,
-                                        discard_regex=discard_regex, iam_role_duration=iam_role_duration,
+                                        only_logs_after=only_logs_after, account_alias=account_alias ,region=region,
```
Comment (GGP1) [inline wodles/aws/services/cloudwatchlogs.py:142]:
```suggestion
                                        only_logs_after=only_logs_after, account_alias=account_alias, region=region, 
```

File: wodles/aws/tests/aws_utils.py [hunk 3]
```diff
@@ -270,7 +270,7 @@ def get_aws_bucket_parameters(db_table_name: str = TEST_TABLE_NAME, bucket: str
 def get_aws_service_parameters(db_table_name: str = TEST_TABLE_NAME, service_name: str = 'cloudwatchlogs',
                                reparse: bool = False, access_key: str = None, secret_key: str = None,
                                profile: str = TEST_AWS_PROFILE, iam_role_arn: str = None,
-                               only_logs_after: str = None, region: str = None, aws_log_groups: str = None,
+                               only_logs_after: str = None, account_alias: str = None ,region: str = None, aws_log_groups: str = None,
```
Comment (GGP1) [inline wodles/aws/tests/aws_utils.py:273]:
```suggestion
                               only_logs_after: str = None, account_alias: str = None, region: str = None, aws_log_groups: str = None,
```

**Review by GGP1 — CHANGES_REQUESTED** [review]

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by Selutario — APPROVED** [review]
LGTM. The key point of the `aws_account_alias` is seeing the alias in the alerts to differentiate the origin account in the dashboard easily. However, after reading this you mentioned [here](https://github.com/wazuh/wazuh/issues/15840#issuecomment-1984089965), I do understand that this is not possible for both Cloudwatch and Inspector services. 

I'm merging it for now, but if this is confusing to users in the future, we may choose to remove it from the code again and also (especially) from the [Services documentation](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/wodle-s3.html#service-aws-account-alias).

**Comment by Selutario** [comment]
The failing checks are unrelated to these changes.

**Pull request merged**